plugins/basic/python/tk_maya_basic/plugin_logic.py

```diff
@@ -33,6 +34,68 @@
 
 logger = sgtk.LogManager.get_logger(__name__)
 
+class ProgressHandler(QtCore.QObject):
+    """
+    An object that wraps a QTimer that is used to periodically check
+    for updates that need to be made to the progress bar in Maya. This
+    will always execute progress updates on the main thread.
+    """
+    PROGRESS_INTERVAL = 150 # milliseconds
+
+    def __init__(self):
+        try:
+            import shiboken2 as shiboken
+        except ImportError:
+            import shiboken
+
+        ptr = OpenMayaUI.MQtUtil.mainWindow()
+        parent = shiboken.wrapInstance(long(ptr), QtGui.QMainWindow)
+
+        super(ProgressHandler, self).__init__(parent=parent)
+
+        self._progress_value = None
+        self._message = None
+        self._timer = QtCore.QTimer(parent=self)
+
+        self._timer.timeout.connect(self._set_progress)
+        self._timer.start(self.PROGRESS_INTERVAL)
+
+    @property
+    def timer(self):
+        """
+        The QTimer instance that's updating progress.
+        """
+        return self._timer
+
+    def _set_progress(self):
```

Comment:
Nitpick: `set_something` without a parameter to set a value feels weird. Maybe `update_progress_bar`?